### PR TITLE
clang-tidy stuff

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: hardware/EvohomeBase.h [hunk 1]
```diff
@@ -395,8 +401,17 @@ class CEvohomeMsg
...
-	CEvohomeMsg(const char * rawmsg) :flags(0), type(pktunk), timestamp(0), command(0), payloadsize(0), readofs(0), enccount(0) { DecodePacket(rawmsg); }
+	CEvohomeMsg() = default;
+	CEvohomeMsg(const char *rawmsg)
+		: flags(0)
+		, timestamp(0)
+		, command(0)
+		, payloadsize(0)
+		, readofs(0)
+		, enccount(0)
+	{
```
Comment (gizmocuz) [inline hardware/EvohomeBase.h:406]: Will all of these not be initialized in the header with the PR ?

Comment (neheb) [reply hardware/EvohomeBase.h:406]: Good catch. Will fix.

**Comment by neheb** [comment]
Rebased to squash commits with identical clang-tidy checks.  Reordered several as well.

**Comment by gizmocuz** [comment]
Thanks for all the hard and many work again !!!

**Pull request merged**